src/main/resources/db/migration/V046__create_type_entity_type_enum.sql

```diff
@@ -0,0 +1,5 @@
+CREATE TYPE entity_type_enum AS ENUM ('PRIVATE', 'PUBLIC_ADMINISTRATION');
+CREATE CAST (character varying AS entity_type_enum) WITH INOUT AS ASSIGNMENT;
+
+ALTER TABLE agreement ADD COLUMN entity_type entity_type_enum;
```

Comment:
No default for existing agreements and profile? Why replicating data in profile and keep it consistent during all operations?

Reply:
Added "PRIVATE" AS DEFAULT